packages/framer-motion/src/animation/animators/js/index.ts

```diff
@@ -280,7 +280,7 @@ export function animateValue<V = number>({
         playState = "finished"
         onComplete && onComplete()
         stopAnimationDriver()
-        updateFinishedPromise()
+        resolveFinishedPromise()
```

Comment:
Could you elaborate a little more on the race condition? Why do we stop updating the finished promise in `finish()`?

Reply:
For reasons slightly beyond my comprehension, but at least related to adding `queueMicrotasks` (which is when Promises are resolved), swapping to a new `Promise` as soon as we resolve the old one messes with `Promise.all` - seemingly at random. As noted in Onne's linked ticket, even adding a `console.log` can affect timings in a way that more consistently fixes the observed bug. So instead we swap to a new `Promise` later - if and when we start a new animation.

Reply:
> swapping to a new Promise as soon as we resolve the old one messes with Promise.all
Would it have the same impact when `cancel()` is called?

Reply:
I haven't see a bug related to this but it's added there because it's been removed from `updateFinishedPromise()`